### PR TITLE
`q` is now completely removed

### PR DESCRIPTION
**Pull request by annando**
One more file is removed from the `include` path.

### CODE REVIEW

File: src/Module/Search/Acl.php [hunk 1]
```diff
@@ -41,11 +41,11 @@
  */
 class Acl extends BaseModule
 {
-	const TYPE_GLOBAL_CONTACT        = 'x';
-	const TYPE_MENTION_CONTACT       = 'c';
+	const TYPE_GLOBAL_CONTACT        = 'x'; //
```
Comment (MrPetovan) [inline src/Module/Search/Acl.php]: ?

File: src/Content/Widget.php [hunk 2]
```diff
@@ -91,9 +91,9 @@ public static function findPeople()
 	}
 
 	/**
-	 * Return unavailable networks
+	 * Return unavailable networks as array
 	 */
-	public static function unavailableNetworks()
+	public static function unavailableNetworksAsArray()
```
Comment (MrPetovan) [inline src/Content/Widget.php:96]:
```suggestion
	public static function unavailableNetworks()
```

File: src/Content/Widget.php [hunk 3]
```diff
@@ -125,6 +125,15 @@ public static function unavailableNetworks()
 		if (!Addon::isEnabled("pnut")) {
 			$networks[] = Protocol::PNUT;
 		}
+		return $networks;
+	}
+
+	/**
+	 * Return unavailable networks
+	 */
+	public static function unavailableNetworks()
```
Comment (MrPetovan) [inline src/Content/Widget.php:134]:
```suggestion
	public static function unavailableNetworksCondition()
```
Why not return a `[condition, parameter]` condition?

File: src/Module/Search/Acl.php [hunk 4]
```diff
@@ -114,76 +114,51 @@ private static function regularContactSearch(string $type)
...
-				intval(local_user())
-			);
-			$group_count = (int) $r[0]['g'];
+			$group_count = DBA::count('group', $condition_group);
 		}
 
-		$sql_extra2 .= ' ' . Widget::unavailableNetworks();
+		$networks = Widget::unavailableNetworksAsArray();
+		if (!empty($networks)) {
+			$condition = DBA::mergeConditions($condition, array_merge(["NOT `network` IN (" . substr(str_repeat("?, ", count($networks)), 0, -2) . ")"], $networks));
```
Comment (MrPetovan) [inline src/Module/Search/Acl.php:135]: Isn’t this the same as `unavailableNetworks`?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Search/Acl.php (hunk 4 above)
Comment (annando) [reply src/Module/Search/Acl.php:135]: `unavailableNetworksAsArray` is returning the networks as array while `unavailableNetworks` is returning a part of a SQL query.

Comment (MrPetovan) [reply src/Module/Search/Acl.php:135]: And this looks like it's returning a part of an SQL query about unavailable networks, the same function as `unavailableNetworks`.

Comment (annando) [reply src/Module/Search/Acl.php:135]: No. The one returns a string with the complete query (that also contains the values) and the new one returns an array with the query with the placeholders and the values in separate array elements.

File: src/Content/Widget.php (hunk 3 above)
Comment (annando) [reply src/Content/Widget.php:134]: I wanted to make this the most generic usable at the moment. We can always change this anytime if needed.

File: src/Module/Search/Acl.php (hunk 4 above)
Comment (MrPetovan) [reply src/Module/Search/Acl.php:135]: Why not make the `unavailableNetworks` method return the SQL condition array replace this line with it ? Currently it looks like the same feature is duplicated in two very separate locations.

Comment (annando) [reply src/Module/Search/Acl.php:135]: The function is used at several places. In a next iteration we can have a look at the other places as well.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**